### ISSUE TITLE
Open Grant Proposal, Noosphere Protocol Adoption

### ISSUE THREAD

**Issue by danversfleury** [filecoin-project/devgrants#1268]
# Open Grant Proposal: `Noosphere Protocol Adoption`

**Name of Project:** Noosphere Protocol Adoption

**Proposal Category:** `docs`

**Proposer:** `danversfleury`

**Technical Sponsor:** Dietrich Ayala (@autonome)

**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT, APACHE2, or GPL licenses?:** Yes

# Project Description

Our species must tackle existential challenges such as the rise of unaligned AI, climate change and the decline of global social mobility. To successfully navigate these challenges we need tools that enable more effective collaboration. We believe a federated commons that supports both permanence and versioning, built on the foundation of peer-to-peer linking, would lead to a new age of interoperability for collaboration tools. Unfortunately, today’s Internet protocols can’t enable this (see Gordon Brander's excellent summaries of this topic [here](https://subconscious.substack.com/p/thinking-together) and [here](https://subconscious.substack.com/p/weblike-things)).

Protocols being developed within the FileCoin and Protocol Labs community could be the solution. But we’ll never know without wide, **rapid protocol adoption**. To achieve this we feel the foundational question to be answered is socio-technical in nature: how do we **engage and train developers while helping them overcome resistance to change?**

In order to accelerate developer adoption, we plan to **create interactive discovery and tutorial experiences that foster confidence, familiarity and competence with a new protocol.** Specifically, we propose co-developing a set of learning experiences with developers implementing [the Noosphere protocol](https://github.com/subconsciousnetwork/noosphere). Why Noosphere?

1. Noosphere is a decentralized protocol that leverages several core FileCoin ecosystem technologies: specifically, Noosphere uses the IPLD format while baking in UCANs at the core for authorization and syndicating content via IPFS.
2. Noosphere solves a critical problem for a specific set of applications: interoperability between "Tools for Thought" applications. As such, it's already applicable to a wide variety of use cases.
3. The grant's principal investigators are well-connected with "Tools for Thought" creators from which we can source potential implementers. 

As we co-develop this resource within the TfT community, the interactive experience could take shape as [interactive explorable explanations](https://explorabl.es/all/), digital [Reggio-inspired innovation spaces](https://www.smore.com/4hg1m-creating-innovative-learning-spaces) for adults, or build-your-own-noosphere-implementation kits. The interactive components will be built on top of Noosphere itself, allowing us to give real-world feedback to the Noosphere core team while also providing reference implementations for future implementers.

## Value

**Benefits to getting it right:**
If we develop an effective learning experience, there are several potential benefits. On a small scale, we hope to see increased adoption of the Noosphere protocol within the "Tools for Thought" community. Noosphere leverages IPFS for long-term archiving & syndication, so increased usage of Noosphere increases IPFS usage. On a slightly larger scale, if this approach is successful, we could share a blueprint for accelerating the adoption of other protocols within the FileCoin and Protocol Labs community. And on the largest scale, if we can facilitate widespread adoption of protocols that enable more effective collaboration, it could help humanity tackle some of the most pressing challenges we face, such as unaligned AI, climate change, and declining social mobility.

**Risks if we don't get it right:**

- **Brand Damage:** We turn people off to Noosphere, which affects Subconscious, the product’s adoption, causing the Subconscious team to fail.
- **Distraction:** We overwhelm Noosphere with users, distracting the core team from developing their Subconscious product.
- **Partial Success as Failure:** By getting adopters through a difficult part of onboarding, we may wind up ‘beaching’ them at another choke point that we aren’t able to address.

**Executional risk:**
The biggest project execution risk is that Noosphere is nascent, which means it is also not yet stable. Breaking changes could create difficulty of adoption issues which could be out of our control to immediately impact. While it is designed as a permissionless, interoperable protocol in the spirit of SMTP or HTTP, there are no applications other than the Subconscious team’s own tool currently implementing it. The protocol itself could turn out to be insufficient to enable collaboration tooling. However, in the process of building learning experiences for Noosphere, we will be acting as a “second system” integrating with Noosphere, allowing us to shape and improve the protocol as we work with it.

## Deliverables

The primary deliverable for this grant will be an interactive tutorial experience for the Noosphere Protocol. This tutorial will be designed to explain and demonstrate the most important concepts underlying Noosphere, as determined through interviews with potential Noosphere implementers.

An interactive tutorial experience is a learning tool that utilizes various methods to engage and educate developers on a particular technology or concept, incorporating elements such as hands-on exercises, visualizations, and gamification to help developers understand and internalize the material. Interactive tutorials can take many forms including:

- Giving users an experience of using the technology in a working prototype with explanations at each step, as in [passkeys.io](https://www.passkeys.io/).
- Visualizing a difficult-to-grasp concept, as in Bartosz Ciechanowski’s [explanation of GPS](https://ciechanow.ski/gps/).
- Gamifying the learning progression by dividing the experience into micro-lessons that can be individually completed, as in [CryptoZombies](https://cryptozombies.io/).
- Applying a spaced-repetition approach to the learning progression, as in [Execute Program](https://www.executeprogram.com/).

The exact form and content will be informed by interviews and iterative testing with potential Noosphere implementers. Interactive elements will be built around a working proof-of-concept of a Noosphere-powered PKM app to provide hands-on experience with the protocol. Additionally, the experience will suggest possible UX patterns for Noosphere-specific features, while providing references for related work and open problems.

## Development Roadmap

The four stages of this project are Discover, Iteratively Prototype, Launch, and Maintenance & Upgrade. We anticipate spending 16 weeks total on the first three and a minimum of two years on the fourth.

### Milestone 1: Discovery

Discovery will reveal the most important blockers to protocol adoption and provide hypotheses to test for solutions to those blockers.

Deliverables:

- Create an initial Adopter Hypotheses
- Launch an Operational Noosphere Network

Dates (4 weeks): 1/16-2/17
People: Danvers + Jess
Budget: $27,000

We will conduct user interviews with 10 to 30 potential Noosphere users and implementers to determine how best to help them adopt Noosphere using a modified version of [Steve Blank-style Customer Development](https://steveblank.com/category/customer-development/). We will also conduct user interviews with 10 to 30 users who have adopted and stuck with a new protocol within the last 10 years in service of understanding their [Jobs to be Done](https://valchanova.me/customer-development-jobs-to-be-done/).

This primary research will be combined with industry best-practices on optimizing software developer experience ([DevEx](https://www.protocol.com/braintrust/developer-experience-hardest-challenges)), past protocol adoption success, and winning adoption wars in general ([think Betamax vs. VHS](https://legacybox.com/blogs/analog/vhs-beat-betamax#:~:text=When%20Betamax%20hit%20the%20market,for%20recording%20and%20watching%20movies.)) to frame out our initial hypothesis on what matters most and why.

The Noosphere protocol currently requires a self-hosted and maintained network in order to adopt. We will set up and maintain our own Spheres and become familiar with the Noosphere tech stack. As part of this effort, we will improve the Noosphere libraries and documentation as needed.

### Milestone 2: Iteratively Prototype

Iterative Prototyping will determine the shape and scope of learning experience as well as some leading indicators on their preferences for experience delivery medium.

Deliverable: A prototyped solution built off of learning from the Discovery phase
Dates (8 weeks): 2/17 - 4/21
People: Danvers + Jess
Budget: $54,000

For the second milestone of our project, we will create a prototype of our learning experience for the Noosphere Protocol. This will include explanations of the most important concepts underlying Noosphere, as determined through interviews with potential implementers. As different sections are ready to share, we will test the content with a group of alpha testers to gather feedback. Based on this feedback, we will iterate on the prototype to improve its efficacy.

The learning experience will include interactive examples of a Noosphere-integrated PKM app, as well as explorations of possible UX for Noosphere-specific features. We may also build an actual plugin for an existing TFT to further demonstrate the value of Noosphere to developers and users.

Our tests will be designed to inform:

- Cohesion: Is the content correct? Does the content “flow” logically? Are the topics discussed clear and complete?
- Interactivity: is an example interactive experience attractive, understandable and sufficient?

### Milestone 3: Launch

We will progressively launch the experience, gathering feedback, and sharing our learnings.

Deliverables:

- An open source protocol launch playbook to be utilized by the IPFS community and beyond.
- Publicly posted results of the experiments associated with the Launch phase.

Dates (4 weeks): 4/21-5/19
People: Danvers + Jess
Budget: $27,000

For the third milestone of our project, we will launch the interactive tutorial for the Noosphere Protocol. This strategy will include a number of different tactics, such as sharing the tutorial on ProductHunt and Hacker News, hosting a Tools for Thought Rocks group session to demonstrate the tutorial, and sharing it with other related communities. Additionally, we will create a YouTube video to showcase the tutorial and highlight its key features. Based on the experience and learning from this launch we will put together a shareable, open-source playbook for other protocols to learn from.

Throughout the development of the tutorial, we will also be building an email list of interested users and stakeholders. When the tutorial is ready for launch, we will email this list to let them know about the availability of the tutorial and encourage them to try it out. This will help to drive adoption of the Noosphere Protocol by demonstrating the value of the protocol to TFT developers and users.

### Milestone 4: Maintenance & Upgrade

Maintenance Dates: 5/19/23 - 12/31/24
People: Danvers & TBD Maintainer
Budget: $18,000 ($6k for optimization + $12k for maintenance)

During the final milestone of our project, we will run a series of final content optimization experiments to improve throughput. This will involve testing different versions of the tutorial and its various components to see which ones are most effective at engaging users and helping them learn the material. Based on the results of these experiments, we will make any necessary changes to the tutorial to improve its performance and make it more effective at driving the adoption of the Noosphere Protocol.

In addition to optimizing the tutorial, we will focus on maintaining the interactive tutorial. This will include keeping the tutorial up and running, and ensuring that it is available for users to access and learn from. We will leverage funds to source and fund a developer from the community to keep up with routine maintenance through at least the end of 2023.

## Total Budget Requested

**Total Budget: $126,000**

Here's the budget broken out by phases and dates:

| Phase                 | Dates              | Budget   |
|-----------------------|--------------------|----------|
| Discover              | 1/16 - 2/17        | $27,000  |
| Prototype             | 2/17 - 4/21        | $54,000  |
| Launch                | 4/21-5/19          | $27,000  |
| Maintenance & Upgrade | 5/19/23 - 12/31/24 | $18,000  |
| Total:                | 1/16/23 - 12/31/24 | $126,000 |

# Team

## Team Members

- Danvers Fleury
- Jess Martin

## Team Member LinkedIn Profiles

- Danvers Fleury: https://www.linkedin.com/in/danversfleury/
- Jess Martin: https://www.linkedin.com/in/jessmartin/

## Team Website

[SocioTechnica.org](https://sociotechnica.org/)

## Relevant Experience

**Danvers Fleury, The Experience Designer**

I am presently an independent consultant where I design and conduct market explorations and user discovery sessions for startups seeking to improve product market fit. I was recently Founder/CEO of a [Techstars backed adult learning app](https://vimeo.com/user44471209/review/366747074/d1f024712d) that utilized neuroscience and behavioral economics to drive workplace application of new skills at a 15X+ rate to conventional training as explained in Ch. 9 of Nate Andorsky’s _[Decoding the Why](https://drive.google.com/file/d/15L_3kL5YaR-V1sKKfE8IhYeA1zsvF_wi/view?usp=share_link)_. Previously, I ran a [digital agency in the higher education space](https://www.verifiedstudios.com/case_studies/nc-state-multisite-case-study/) and oversaw product marketing at three acquired startups.

I am deeply familiar with creating educational technology environments that get people to try new things, overcome emotional barriers and then commit to real change and action. I have also designed and optimized the growth funnels of multiple products by developing nuanced customer insights and matching messaging and user experience to deep user needs. I believe these two skill sets are uniquely suited to the success of building an interactive learning environment that drives protocol adoption for Noosphere.

**Jess Martin, The Subject Matter Expert + Developer**

I am currently an engineer at Fission where I am in charge of exploring applications enabled by Fission technology and driving adoption of those technologies. I have two decades of software development experience, primarily focused on early-stage startups, where I have done everything from full-stack engineering to acting as CTO. I’m also a co-organizer of [Tools for Thought Rocks](https://toolsforthought.rocks/), a 500 person meetup of TfT creators.

Throughout my career, I’ve built prototypes of dozens of software programs, focusing on first-time usage and accelerating adoption. Upon discovering Noosphere, I built [an interactive Noosphere Simulator](https://github.com/jessmartin/noosphere-simulator) in order to teach myself the basic concepts of the protocol. I enjoy explaining deeply technical concepts to the average developer. You can watch livestreams of me explaining and re-implementing Noosphere [here](https://www.youtube.com/watch?v=63tLxVWN4Ao) and [here](https://www.youtube.com/watch?v=YkD5mN0Qh2s). My experience as an early-stage developer, my understanding of the Noosphere protocol, and my penchant for explaining technical concepts in layman’s terms should be well suited to crafting an interactive learning environment that drives adoption for Noosphere.

**Team Experience**

Together we have worked on close to a dozen projects since 2008. Jess was a key team member of a startup incubator that Danvers co-founded, Danvers acted as the lead consultant for designing and developing a tech startup that Jess co-founded that was [acquired by RE/MAX](https://www.prnewswire.com/news-releases/remax-acquires-data-science-startup-first-continuing-the-brands-technological-transformation-300977132.html), and we recently teamed up to architect and launch [Bloomtech’s Backend Development](https://www.bloomtech.com/courses/backend-development) course, which presently boasts the institution’s highest placement rates.

Our latest joint efforts are published on [Sociotechnica.org](https://sociotechnica.org), where we also share our individual user’s guides ([JM](https://jessmart.in/notes/AE937E9C-B9EC-436E-B7AF-194938CB6D42-51070-000091B125678D86), [DF](https://docs.google.com/document/d/1kSPZf21__PSyvI_UxK-kCEnHI23VOOhw9MBPb7wB-Lg)) and [Most Excellent Agreement](https://docs.google.com/document/d/1VAGkXQopFVlx1VRPfs3IUg22M7mTslLKqjQuqYefRo8/) for how we work together and with others.

## Team code repositories

[Noosphere Simulator](https://github.com/jessmartin/noosphere-simulator)

# Additional Information

As Noosphere co-founder [Chris Joel](https://www.linkedin.com/in/christopherjoel/) put it:

> ...the thought has crossed my mind that this project seems super useful beyond Noosphere. There is a fleet of new technologies that everyone is going to try to deploy over the coming years, and for the most part those technologies are not legible to developers (and especially not “lay” users).

We see documentation as a core area to disrupt and evolve in the protocol space and appreciate your prospective support, to begin to do so with Noosphere!

**Best email:** `danversfleury@gmail.com`

**Comment by ErinOCon** [comment]
Hi @danversfleury, thank you for your proposal and for your patience with our review. We will not be moving forward with a grant for this project. Wishing you the best as you continue to build!
